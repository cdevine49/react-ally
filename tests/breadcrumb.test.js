import React from 'react';
import { cleanup, render } from 'react-testing-library';
import { Breadcrumb, BreadcrumbLink } from '../es/react-ally.js';

afterEach(cleanup);

test('renders correctly', () => {
  const component = (
    <Breadcrumb aria-label="Override me" id="nav-id" listProps={{ id: 'list-id' }}>
      <BreadcrumbLink id="root-id" href="/root">
        Root
      </BreadcrumbLink>
      <BreadcrumbLink href="/root/middle" listItemProps={{ id: 'middle-li' }}>
        Middle
      </BreadcrumbLink>
      <div>Don't show me</div>
      Ignore Me
      <BreadcrumbLink href="/root/middle/current">Current</BreadcrumbLink>
    </Breadcrumb>
  );

  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});
