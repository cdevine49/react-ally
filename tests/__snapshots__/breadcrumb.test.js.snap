// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<DocumentFragment>
  <nav
    aria-label="Breadcrumb"
    id="nav-id"
  >
    <ol
      id="list-id"
    >
      <li>
        <a
          href="/root"
          id="root-id"
        >
          Root
        </a>
      </li>
      <li
        id="middle-li"
      >
        <a
          href="/root/middle"
        >
          Middle
        </a>
      </li>
      <li>
        <a
          aria-current="page"
          href="/root/middle/current"
        >
          Current
        </a>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
